### ISSUE TITLE
pseudohash slices

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#3457]
Migrated from [rt.perl.org#5874](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=5874) (status was 'resolved')

Searchable as RT5874$

**Comment by p5pRT** [comment]
### From me-01@ton.iguana.be



This works&#8203;:
perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo bar\)\} = \(1\,2\); print Dumper\($var\)'

This fails&#8203;:
perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo bar\)\} = \(1\.\.2\); print Dumper\($var\)'

This fails&#8203;:
perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo\)\} = \(1\); print Dumper\($var\)'

This fails&#8203;:
perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;\{$var\}\{qw\(foo bar\)\} = \(1\.\.2\); print Dumper\($var\)'

For the last two I suspect it's just not supported in the parser \(but I 
really think it should be\)\, the first one might indicate a real internal
bug\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.0:

Configured by ton at Wed Aug  9 23:24:07 CEST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.0-test5, archname=i686-linux
    uname='linux quasar 2.4.0-test5 #2 sat jul 29 04:45:14 cest 2000 i686 unknown '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O2 -fomit-frame-pointer', gccversion=egcs-2.91.66 19990314 (egcs-1.1.2 release), gccosandvers=
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccflags ='-fno-strict-aliasing -I/usr/local/include'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib -L/opt/gnu/lib'
    libpth=/usr/local/lib /opt/gnu/lib /lib /usr/lib
    libs=-lnsl -lndbm -ldl -lm -lc -lposix -lcrypt -lutil
    libc=/lib/libc-2.1.1.so, so=so, useshrplib=true, libperl=libperl.so
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/usr/lib/perl5/5.6.0/i686-linux/CORE'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib -L/opt/gnu/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/lib/perl5/5.6.0/i686-linux
    /usr/lib/perl5/5.6.0
    /usr/lib/perl5/site_perl/5.6.0/i686-linux
    /usr/lib/perl5/site_perl/5.6.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/ton
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/ton/bin.Linux:/home/ton/bin:/home/ton/bin.SampleSetup:/usr/local/bin:/usr/local/sbin:/usr/local/jre/bin:/usr/local/jdk116_v5/bin:/home/oracle/product/8.1.5/bin:/usr/games/bin:/usr/X11R6/bin:/usr/share/bin:/usr/bin:/usr/sbin:/bin:/sbin:.
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

In article \<20010220181536\.8500\.qmail@&#8203;quasar\.home\.lunix>\,
&nbsp;    me\-01@&#8203;ton\.iguana\.be writes&#8203;:

Following up to myself\.\.

> This is a bug report for perl from thospel@&#8203;mail\.dma\.be\,
> generated with the help of perlbug 1\.31 running under perl v5\.6\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> This works&#8203;:
> perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo bar\)\} = \(1\,2\); print Dumper\($var\)'
> 
> This fails&#8203;:
> perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo bar\)\} = \(1\.\.2\); print Dumper\($var\)'

ANY use of list context \.\. anywhere before using a lexical variable
seems to make them not work\. E\.g&#8203;:
Works&#8203;: perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; $var\->\{foo\} = 1; print Dumper\($var\)'
Fails&#8203;: perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; my @&#8203;a=1\.\.2; $var\->\{foo\} = 1; print Dumper\($var\)'

> This fails&#8203;:
> perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;$var\{qw\(foo\)\} = \(1\); print Dumper\($var\)'
> 
> This fails&#8203;:
> perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;\{$var\}\{qw\(foo bar\)\} = \(1\.\.2\); print Dumper\($var\)'

Typo\, in case you think this is just the list \.\. case\.
This fails too&#8203;:
perl \-wle 'use Data&#8203;::Dumper; use fields qw\(foo bar\); my main $var=\[\]; @&#8203;\{$var\}\{qw\(foo bar\)\} = \(1\,2\); print Dumper\($var\)'

> 
> For the last two I suspect it's just not supported in the parser \(but I 
> really think it should be\)\, the first one might indicate a real internal
> bug\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

I suspect someone saw "pseudohash" in the subject\, and thought&#8203;:
"we are dropping pseudo\-hashes\, so any bugs are not relevant\.

However\, I choose that subject name before the time pseudohashes
were decided to be dropped\, and it was a badly chosen one\.

The bug is really about "use fields" and the way perl parses the
hash like notation of fields\. As far as I know\, the set of problems 
\(most notational\, 1 certainly a bug\) is UNSOLVED\.
\(I base myself on what I read in p5p\. I didn't download and try the latest
patches\, so ignore all of this mail if it's really solved in bleadperl\)

So\, unless I'm mistaken and it \*IS\* solved I would ask&#8203;:

1\) Resurrect the bug report
2\) Introduce a rule that a bug report can never be closed without stating
&nbsp;   a reason\.
3\) I'd really like it if the "closer" of a bug was visible\, so it's 
&nbsp;   possible to send a mail to ask or disagree with the closing\.
4\) NEVER close a bug unless you read the report and thought about the
&nbsp;   contents\. The subject is \*NOT\* enough\.

**Comment by p5pRT** [comment]
### From @rspier

&nbsp;    I'm CC'ing Richard\, and hopefully he can figure out what happened\.

\-R

Ton Hospel writes&#8203;:
>I just received a message that bugid 20010220\.007 was closed\.
>\(see http&#8203;://bugs\.perl\.org/perlbug\.cgi?req=bug\_id&bug\_id=20010220\.007\)
>However\, nowhere can i find the reason WHY it was closed\.
>
>I suspect someone saw "pseudohash" in the subject\, and thought&#8203;:
>"we are dropping pseudo\-hashes\, so any bugs are not relevant\.
>
>However\, I choose that subject name before the time pseudohashes
>were decided to be dropped\, and it was a badly chosen one\.
>
>The bug is really about "use fields" and the way perl parses the
>hash like notation of fields\. As far as I know\, the set of problems 
>\(most notational\, 1 certainly a bug\) is UNSOLVED\.
>\(I base myself on what I read in p5p\. I didn't download and try the latest
>patches\, so ignore all of this mail if it's really solved in bleadperl\)
>
>So\, unless I'm mistaken and it \*IS\* solved I would ask&#8203;:
>
>1\) Resurrect the bug report
>2\) Introduce a rule that a bug report can never be closed without stating
>   a reason\.
>3\) I'd really like it if the "closer" of a bug was visible\, so it's 
>   possible to send a mail to ask or disagree with the closing\.
>4\) NEVER close a bug unless you read the report and thought about the
>   contents\. The subject is \*NOT\* enough\.

**Comment by p5pRT** [comment]
### From @nwc10

> Ton Hospel writes&#8203;:
> >So\, unless I'm mistaken and it \*IS\* solved I would ask&#8203;:
> >
> >1\) Resurrect the bug report
> >2\) Introduce a rule that a bug report can never be closed without stating
> >   a reason\.
> >3\) I'd really like it if the "closer" of a bug was visible\, so it's 
> >   possible to send a mail to ask or disagree with the closing\.
> >4\) NEVER close a bug unless you read the report and thought about the
> >   contents\. The subject is \*NOT\* enough\.


I would agree\. I confess that I've been a very bad bug admin\, and particularly
bad at closing my own bugs\, and independent of that wasn't going to mail it
here\, but 20010303\.010 has been closed\, probably based on the subject\.
\[warnings not being UTF8 aware\]
However\, there was a second bug in there about different behaviour of
unary minus for strings in the range 128\-255 depending on whether they
happen to be UTF8 encoded or not which is still live\. I should resubmit
that so that it gets a pertinent subject\.

However of Ton's thoughts\, I'd agree with all 4 for my bug too \(if I'm
allowed to change the subject\)

Is Richard's mail working again yet? It's hard to contact him when
550 relaying denied is the order of the day\.

Nicholas Clark

**Comment by p5pRT** [comment]
### From @rspier

That should definitely be open\.  It\, like 20010220\.007\, doesn't have
an administrator listed\.  Something weird is going on\.

>However of Ton's thoughts\, I'd agree with all 4 for my bug too \(if I'm
>allowed to change the subject\)

> 1\) Resurrect the bug report

Yes\, but lets wait until Richard speaks\.  We don't want to overwrite
something in the database\.

> >2\) Introduce a rule that a bug report can never be closed without stating
> >   a reason\.

I generally agree with this\, and try and put notes in the database
when it isn't obvious from the last message in the thread\.

> >3\) I'd really like it if the "closer" of a bug was visible\, so it's 
> >   possible to send a mail to ask or disagree with the closing\.

It generally is\.

> >4\) NEVER close a bug unless you read the report and thought about the
> >   contents\. The subject is \*NOT\* enough\.

Duh\.  :\)

>Is Richard's mail working again yet? It's hard to contact him when
>550 relaying denied is the order of the day\.

Yes\.
&nbsp;    richard@&#8203;perl\.org or  Richard\.Foley@&#8203;t\-online\.de

should get to him\.

\-R

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

I saw this yesterday with a couple of my bugs\, but hadn't got around
to working out what was going on\.

Michael

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

> >     I'm CC'ing Richard\, and hopefully he can figure out what happened\.
Hello \- I closed a few bugs myself recently\, mostly based on the chain
of replies\, it's possible this was one of them\, in which case it sounds
as though I may have been over zealous\.

> > Ton Hospel writes&#8203;:
> > >So\, unless I'm mistaken and it \*IS\* solved I would ask&#8203;:
> > >
> > >1\) Resurrect the bug report
Done

> > >2\) Introduce a rule that a bug report can never be closed without stating
> > >   a reason\.

In a perfect world\.  However\, I've closed bugs without stating a reason\,
but only where I felt the chain of connecting emails was clear enough to
determine that the bug either wasn't\, or it was clearly dealt with\,
patched\,  etc\.  There's a lot of 'bugs' in the db that aren't actually
bugs at all\, I don't think the 904 open bugs are truly representative of
the current status\.

Damn\, it was under 900 yesterday\!

> > >3\) I'd really like it if the "closer" of a bug was visible\, so it's
> > >   possible to send a mail to ask or disagree with the closing\.

Yup\, this should always be the case\.  Currently there are issues with
the History \(don't use it\)\, which I hope to fix shortly\.  Tuits are low
on the ground at the moment however\.  

I spent a lot of time on the API and interface a while ago\, announced
it\, put it on SourceForge \(so you can all have a go\)\, and since then
have struggled to find more time to commit \- apologies\.

There is of course the ever present spectre of the entire software tree
being available by CVS at SourceForge under the Perlbug project\, so feel
free to go hack/fix/replace anytime :\-\)

> > >4\) NEVER close a bug unless you read the report and thought about the
> > >   contents\. The subject is \*NOT\* enough\.
> 
Unless the subject reads&#8203;: 'Sexy Rita Offers Free Viagra\!' 

;\-\)

> I would agree\. I confess that I've been a very bad bug admin\, and particularly

Tut tut

> Is Richard's mail working again yet? 
Yup \- I'm back

> It's hard to contact him when
Nice to be wanted :\-\)

> 550 relaying denied is the order of the day\.
> 
I thought I sent a mail to p5p to say which email addresses to use\, as
soon as my address went awol?  It's OK now\, and looked after by 'real
people' \(thanks Ask\)\, instead of idiots\, so you/I shouldn't have any
more problems with this\.

For future \(emergency\) reference&#8203;:

&nbsp;    \+49 \(0\)8085 189781

**Comment by p5pRT** [comment]
@cwest - Status changed from 'open' to 'resolved'
